### PR TITLE
[SYCL][DOC] Add an alias for empty_property_list

### PR DESCRIPTION
**Pull request by KornevNikita**

### CODE REVIEW

File: sycl/doc/design/CompileTimeProperties.md [hunk 1]
```diff
@@ -45,7 +45,7 @@ for declaring global variables.  One such example is the
 ```
 namespace sycl::ext::oneapi {
 
-template <typename T, typename PropertyListT = properties<>>
+template <typename T, typename PropertyListT = experimental::empty_property_list>
```
Comment (steffenlarsen) [inline sycl/doc/design/CompileTimeProperties.md:48]: I would prefer reducing the ambiguity of the name by making it `empty_properties_t` to both make it clear this is a type alias and to not make it sound like it is related to `sycl::property_list`. Tag @gmlueck @rolandschulz @Pennycook .

On another note, we should also fix this design doc so it doesn't mention experimental extensions in non-experimental namespaces.

Comment (gmlueck) [reply sycl/doc/design/CompileTimeProperties.md:48]: Yes, I also prefer `empty_properties_t`.

> On another note, we should also fix this design doc so it doesn't mention experimental extensions in non-experimental namespaces

Yes, good catch.

File: sycl/doc/extensions/experimental/sycl_ext_oneapi_properties.asciidoc [hunk 2]
```diff
@@ -405,6 +405,8 @@ class properties {
   static constexpr auto get_property();
 };
 
+using empty_property_list = decltype(properties{});
+
```
Comment (gmlueck) [inline sycl/doc/extensions/experimental/sycl_ext_oneapi_properties.asciidoc:409]: Could you also add a sentence below describing this type alias?  For example, add this right after the table of member functions:

> The `empty_properties_t` type alias is a shorthand for the type of an empty property list.

**Review by gmlueck — APPROVED** [review]

**Review by steffenlarsen — APPROVED** [review]
Thanks a ton! 🚀

**Pull request merged**